src/build-data/detect_version.cpp

```diff
@@ -0,0 +1,56 @@
+/*
+* This file is preprocessed to produce output that is examined by
+* configure.py to determine the compilers version number.
+*/
+
+#if defined(_MSC_VER)
+
+// _MSC_VER Defined as an integer literal that encodes the major and minor
+// number elements of the compiler's version number. The major number is
+// the first element of the period-delimited version number and the minor
+// number is the second element. For example, if the version number of the
+// Visual C++ compiler is 17.00.51106.1, the _MSC_VER macro evaluates to 1700.
+// https://msdn.microsoft.com/en-us/library/b0084kay.aspx
+
+MSVC _MSC_VER
+
+#elif defined(__clang__)
+
+#if defined(__apple_build_version__)
+
+/*
+Map Apple XCode versions back to standard Clang
```

Comment:
Did you test what Apple clang prints for `__clang_major__` and `__clang__minor__`? It may be the compiler version like "clang-900.0.39.2" or "clang-900.0.38" which is nit strictly bound to the XCode version

Reply:
No way for me to test locally, but https://stackoverflow.com/questions/19387043/how-can-i-reliably-detect-the-version-of-clang-at-preprocessing-time has examples of the macro for older XCodes. I couldn't find anything that lists the `__clang_major__` and `__clang_minor__` values for recent versions :/